internal/controller/thanosquery_controller.go

```diff
@@ -242,16 +242,6 @@ func (r *ThanosQueryReconciler) getStoreAPIServiceEndpoints(ctx context.Context,
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ThanosQueryReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	servicePredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
```

Comment:
Probably misunderstanding this, but why do we remove the store label filter here, and in the enqueue method?